File: tests/combi/matrix_intro_malloc.cpp
#include <stdlib.h>
#include "../../include/optitrust.h"
const int N = 10;
int* q;

void allocate (){
 q = (int*) malloc(N * sizeof(int));
}

int main () {
  int* p = (int*) malloc (N * sizeof(int));
  return 0;
}

File: tests/combi/matrix_intro_malloc.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun () -> 

  !! Matrix.intro_malloc [cVarDef "p"];

)

"
#include <stdlib.h>
#include \"../../include/optitrust.h\"

int main() {
  const int N = 10;
  int* p = (int*) malloc (N * sizeof(int));
}
"

let _ = Run.script_cpp (fun _ ->

    !! Matrix.intro_malloc [cVarDef "p"];
    !! Matrix.intro_malloc [cVarDef "q"];

)
